modules/serverpod_auth/serverpod_auth_server/test/firebase/firebase_auth_mock.dart

```diff
@@ -0,0 +1,243 @@
+          if (expiresIn != null) 'expires_in': expiresIn.inSeconds,
+          if (expiresAt != null)
+            'expires_at': expiresAt.millisecondsSinceEpoch ~/ 1000
+        },
+      ).toJson(),
+    });
+  }
+}
+
+String generateMockIdToken({
+  required String projectId,
+  required String uid,
+  Map<String, dynamic>? overrides,
+}) {
+  overrides ??= {};
+
+  var claims = {
+    'aud': projectId,
+    'exp':
+        DateTime.now().add(const Duration(hours: 1)).millisecondsSinceEpoch ~/
+            1000,
+    'iss': 'https://securetoken.google.com/$projectId',
+    'sub': uid,
+    'auth_time': DateTime.now().millisecondsSinceEpoch ~/ 1000,
+    ...overrides,
+  };
+
+  var key = getTestJsonWebKey();
+
+  var builder = JsonWebSignatureBuilder()
+    ..jsonContent = claims
+    ..setProtectedHeader('kid', key.keyId)
+    ..addRecipient(
+      key,
+      algorithm: 'RS256',
+    );
+  return builder.build().toCompactSerialization();
+}
+
+Map<String, dynamic> testAccountServiceJson = {
+  'type': 'service_account',
+  'project_id': 'project_id',
+  'private_key_id': 'aaaaaaaaaabbbbbbbbbbccccccccccdddddddddd',
+  'private_key':
+      '-----BEGIN RSA PRIVATE KEY-----\nMIIEpAIBAAKCAQEAwJENcRev+eXZKvhhWLiV3Lz2MvO+naQRHo59g3vaNQnbgyduN/L4krlr\nJ5c6FiikXdtJNb/QrsAHSyJWCu8j3T9CruiwbidGAk2W0RuViTVspjHUTsIHExx9euWM0Uom\nGvYkoqXahdhPL/zViVSJt+Rt8bHLsMvpb8RquTIb9iKY3SMV2tCofNmyCSgVbghq/y7lKORt\nV/IRguWs6R22fbkb0r2MCYoNAbZ9dqnbRIFNZBC7itYtUoTEresRWcyFMh0zfAIJycWOJlVL\nDLqkY2SmIx8u7fuysCg1wcoSZoStuDq02nZEMw1dx8HGzE0hynpHlloRLByuIuOAfMCCYwID\nAQABAoIBADFtihu7TspAO0wSUTpqttzgC/nsIsNn95T2UjVLtyjiDNxPZLUrwq42tdCFur0x\nVW9Z+CK5x6DzXWvltlw8IeKKeF1ZEOBVaFzy+YFXKTz835SROcO1fgdjyrme7lRSShGlmKW/\nGKY+baUNquoDLw5qreXaE0SgMp0jt5ktyYuVxvhLDeV4omw2u6waoGkifsGm8lYivg5l3VR7\nw2IVOvYZTt4BuSYVwOM+qjwaS1vtL7gv0SUjrj85Ja6zERRdFiITDhZw6nsvacr9/+/aut9E\naL/koSSb62g5fntQMEwoT4hRnjPnAedmorM9Rhddh2TB3ZKTBbMN1tUk3fJxOuECgYEA+z6l\neSaAcZ3qvwpntcXSpwwJ0SSmzLTH2RJNf+Ld3eBHiSvLTG53dWB7lJtF4R1KcIwf+KGcOFJv\nsnepzcZBylRvT8RrAAkV0s9OiVm1lXZyaepbLg4GGFJBPi8A6VIAj7zYknToRApdW0s1x/XX\nChewfJDckqsevTMovdbg8YkCgYEAxDYX+3mfvv/opo6HNNY3SfVunM+4vVJL+n8gWZ2w9kz3\nQ9Ub9YbRmI7iQaiVkO5xNuoG1n9bM+3Mnm84aQ1YeNT01YqeyQsipP5Wi+um0PzYTaBw9RO+\n8Gh6992OwlJiRtFk5WjalNWOxY4MU0ImnJwIfKQlUODvLmcixm68NYsCgYEAuAqI3jkk55Vd\nKvotREsX5wP7gPePM+7NYiZ1HNQL4Ab1f/bTojZdTV8Sx6YCR0fUiqMqnE+OBvfkGGBtw22S\nLesx6sWf99Ov58+x4Q0U5dpxL0Lb7d2Z+2Dtp+Z4jXFjNeeI4ae/qG/LOR/b0pE0J5F415ap\n7Mpq5v89vepUtrkCgYAjMXytu4v+q1Ikhc4UmRPDrUUQ1WVSd+9u19yKlnFGTFnRjej86hiw\nH3jPxBhHra0a53EgiilmsBGSnWpl1WH4EmJz5vBCKUAmjgQiBrueIqv9iHiaTNdjsanUyaWw\njyxXfXl2eI80QPXh02+8g1H/pzESgjK7Rg1AqnkfVH9nrwKBgQDJVxKBPTw9pigYMVt9iHrR\niCl9zQVjRMbWiPOc0J56+/5FZYm/AOGl9rfhQ9vGxXZYZiOP5FsNkwt05Y1UoAAH4B4VQwbL\nqod71qOcI0ywgZiIR87CYw40gzRfjWnN+YEEW1qfyoNLilEwJB8iB/T+ZePHGmJ4MmQ/cTn9\nxpdLXA==\n-----END RSA PRIVATE KEY-----\n',
+  'client_email': 'foo@project_id.iam.gserviceaccount.com',
+  'client_id': 'client_id'
+};
+
+JsonWebKey getTestJsonWebKey() {
```

Comment:
Question: This becomes a lot of code to maintain for the tests, do we really need it? Couldn't we just have a couple of these saved down? Or does that become a problem in some way?

For Example the `intToBase64` method seems strange to implement just for tests.